applications/GeoMechanicsApplication/tests/cpp_tests/test_U_Pw_normal_lysmer_absorbing_condition.cpp

```diff
@@ -0,0 +1,264 @@
+            else if (rVariable == CONFINED_STIFFNESS)
+            {
+                double confined_stiffness = 4.0;
+                rOutput[0] = confined_stiffness;
+                rOutput[1] = confined_stiffness;
+                rOutput[2] = confined_stiffness;
+                rOutput[3] = confined_stiffness;
+
+            }
+            else if (rVariable == DEGREE_OF_SATURATION)
+            {
+                double degree_of_saturation = 0.5;
+                rOutput[0] = degree_of_saturation;
+                rOutput[1] = degree_of_saturation;
+                rOutput[2] = degree_of_saturation;
+                rOutput[3] = degree_of_saturation;
+            }
+            else
+            {
+                Element::CalculateOnIntegrationPoints(rVariable, rOutput, rCurrentProcessInfo);
+            }
+
+        };
+    };
+
+
+            
+        ModelPart::ConditionType::Pointer SetUpUPwLysmerAbsorbingCondition2D2NCondition(ModelPart& rModelPart)
+        {
+            // Set the element properties
+            auto cond_prop = rModelPart.CreateNewProperties(0);
+
+            // Create the test piping element nodes
+            auto node_1 = rModelPart.CreateNewNode(1, 0.0, 0.0, 0.0);
+            auto node_2 = rModelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
+
+            // Create the test Lysmer condition
+            std::vector<ModelPart::IndexType> cond_nodes{ 1, 2 };
+            ModelPart::ConditionType::Pointer cond = rModelPart.CreateNewCondition("UPwLysmerAbsorbingCondition2D2N", 1, cond_nodes, cond_prop);
+
+            // set properties of the condition
+            Vector absorbing_factors = ZeroVector(2);
+            absorbing_factors[0] = 1.0;
+            absorbing_factors[1] = 1.0;
+
+            cond->SetValue(ABSORBING_FACTORS, absorbing_factors);
+            cond->SetValue(VIRTUAL_THICKNESS, 100.0);
+
+            // create mock neighbour element
+            auto p_neighbour_prop = rModelPart.CreateNewProperties(1);
+            p_neighbour_prop->SetValue(POROSITY, 1.0);
+            p_neighbour_prop->SetValue(DENSITY_WATER, 1000);
+            p_neighbour_prop->SetValue(DENSITY_SOLID, 2000);
+
+            auto neighbour_node_1 = rModelPart.CreateNewNode(3, 1.0, 1.0, 0.0);
+            auto neighbour_node_2 = rModelPart.CreateNewNode(4, 0.0, 1.0, 0.0);
+
+            // register mock element
+            const MockElement mock_element{
+                0, Kratos::make_shared<Quadrilateral2D4<Element::NodeType>>(Element::GeometryType::PointsArrayType(4)), Element::PropertiesType::Pointer(0)};
+            KRATOS_REGISTER_ELEMENT("MockElement", mock_element)
+
+            // add neighbour element to model part
+            Element::GeometryType::Pointer p_geometry = Kratos::make_shared<Kratos::Quadrilateral2D4<Element::NodeType>>(node_1, node_2, neighbour_node_1, neighbour_node_2);
+            auto neighbour_element = rModelPart.CreateNewElement("MockElement", 1, p_geometry, p_neighbour_prop);
```

Comment:
Since we create the mock element here anyway, I would suggest to skip the registry and add the element to the modelpart, something like this (I didn't check it in the IDE so there might be typos):
 ```suggestion
            Element::GeometryType::Pointer p_geometry = Kratos::make_shared<Kratos::Quadrilateral2D4<Element::NodeType>>(node_1, node_2, neighbour_node_1, neighbour_node_2);
            const auto mock_element = Kratos::make_intrusive<MockElement>(
                1, p_geometry , Element::PropertiesType::Pointer(0), p_neighbour_prop);
            rModelPart.AddElement(mock_element) 
```